[PICS-Removal] Removing automatable PICS from ICDM 2 1 test module

j-ororke opened the pull request:
- Replaced automatable PICS from TC_ICDM_2_1 test module with checking attributes against supported attribute list on endpoint.

### Testing

j-ororke committed:
Updated TC_ICDM_2_1 test module:
- Removed automatable PICS from TC_ICDM_2_1 test module

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

j-ororke committed:
updated TC_ICDM_2_1: Attempting to resolve issue in CI/CD pipeline

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

j-ororke committed:
Updated TC_ICDM_2_1 test module:
- Enabling verbose output in CI/CD pipeline to get more context on current issue

j-ororke committed:
Updating TC_ICDM_2_1 and test_TC_ICDM_2_1 test modules:
- Removed additional print statements from TC_ICDM_2_1 due to no longer needed
- Added AttributeList to list of attrs in test_TC_ICDM_2_1 to see if we can get the CI/CD to pass.

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

cecille commented on src/python_testing/TC_ICDM_2_1.py:
```diff
@@ -274,8 +274,9 @@ async def test_TC_ICDM_2_1(self):
+
         self.step(11)
-        if self.pics_guard(self.check_pics("ICDM.S.A0009")):
+        if attributes.MaximumCheckInBackOff.attribute_id in attribute_list:
```
Same comment here as on the opstate tests - this gets rid of the skip marker. Though this test is already a mixed bag of whether that happens.

j-ororke replied:
Hi Cecille,
I have created a function in MatterBaseTest class named "attributes_guard" as you had mentioned using the check_attributes function in the matter_testing_scripts helper module. 
This should work I hope, please see the update to the issue [#370](https://github.com/project-chip/matter-test-scripts/issues/370) you have filed and let me know if this will resolve the above comment. 
Thank you!

cecille replied:
is that in this PR or a different PR?

j-ororke replied:
Hi Cecille,
Sorry, the above created attributes_guard function is actually implemented currently in PR #[34290](https://github.com/project-chip/connectedhomeip/pull/34290) for the Opstate PICS replacement tests.

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

j-ororke committed:
Update TC_ICDM_2_1 test module:
- Updating to using attribute_guard()

cecille commented on src/python_testing/TC_ICDM_2_1.py:
```diff
@@ -194,7 +194,7 @@ async def test_TC_ICDM_2_1(self):
         # Validate ClientsSupportedPerFabric
         self.step(5)
-        if self.pics_guard(self.check_pics("ICDM.S.A0005")):
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.ClientsSupportedPerFabric):
```
thinking it makes more sense to have this be a non-async function. See _async_runner for the wrapper on how to do that. I don't think you're ever going to want this to be truly async

j-ororke commented on src/python_testing/TC_ICDM_2_1.py:
```diff
@@ -148,8 +149,7 @@ async def test_TC_ICDM_2_1(self):
         self.step(2)
-        if self.check_pics("ICDM.S.A0002"):
-
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.ActiveModeThreshold):
```
This has been removed in the latest code!

j-ororke commented on src/python_testing/TC_ICDM_2_1.py:
```diff
@@ -178,7 +178,7 @@ async def test_TC_ICDM_2_1(self):
         # Validate IdleModeDuration
         self.step(4)
-        if self.check_pics("ICDM.S.A0000"):
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.IdleModeDuration):
```
This has been removed in the latest code changes. thank you!

j-ororke committed:
Updating TC_ICDM_2_1 test module:
- Updating to remove unneeded if statements for checking mandatory attributes

j-ororke committed:
Updating TC_ICDM_2_1 and matter_testing support module:
- Added new function "async_function_runner" to matter_testing support module,
allowing async functions to be run synchronously even in the event that another event loop is already running
- Updated TC_ICDM_2_1 to using async_function_runner for calling functions such as attribute_guard when not wanted for those to be awaited.

j-ororke committed:
Merge remote-tracking branch 'upstream/master' into remove_PICS_from_ICDM_tests

j-ororke committed:
Updating TC_ICDM_2_1 test module:
- Resolving linting error

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

j-ororke committed:
Updating TC_ICDM_2_1 test module:
- Replaced PICS with new async_function_runner function to resolve issue with unit test failing

j-ororke committed:
Merge branch 'master' into remove_PICS_from_ICDM_tests
